debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerQueryBuilder.java

```diff
@@ -144,7 +144,13 @@ private static String getOperationCodePredicate(OracleConnectorConfig connectorC
             operationInClause.withValues(OPERATION_CODES_LOB);
         }
         else {
-            operationInClause.withValues(OPERATION_CODES_NO_LOB);
+            final List<Integer> operationCodes = new ArrayList<>(OPERATION_CODES_NO_LOB);
+            // The transaction start event needs to be handled when a persistent buffer (Infinispan) is used
+            // because it is needed to reset the event id counter when re-mining transaction events.
+            if (connectorConfig.getLogMiningBufferType() == OracleConnectorConfig.LogMiningBufferType.MEMORY) {
+                operationCodes.removeIf(operationCode -> operationCode == 6);
+            }
+            operationInClause.withValues(operationCodes);
```

Comment:
So IIRC, the connector works the same regardless if Infinispan or Memory-based buffers are used, we always restart the mining process on connector restarts from the offset's `scn` value.  So theoretically, the fact that Infinispan is persistent is primarily irrelevant in that case.

This raises a good question as to whether we should technically just treat Infinispan caches as off-heap data stores that do not persist across restarts.  If we did that, this would, therefore, be identical, no?

Reply:
Yes, I believe that then the behavior will be identical and there will be no need to process the transaction start event.

Reply:
Ok let me check with the one user I know that heavily relies on ISPN.